### PR TITLE
Fail Deprecated Xamarin Test Cloud Task

### PR DESCRIPTION
**Pull request by Pranshu-Negi**
**Task name**: XamarinTestCloudV1

**Description**: Fail Deprecated Tasks with removal date 31-01-2024 owned by Agent and Tasks team

**Documentation changes required:** (N

**Added unit tests:** (Y/N) 

**Attached related issue:** https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/2125781

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/XamarinTestCloudV1/task.loc.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   "version": {
     "Major": 1,
     "Minor": 234,
```
Comment (DmitriiBobreshev) [inline Tasks/XamarinTestCloudV1/task.loc.json:16]: Could you please bump the version of the tasks to 235 since the sprint is almost over?

Comment (Pranshu-Negi) [reply Tasks/XamarinTestCloudV1/task.loc.json]: Done

**Review by DmitriiBobreshev — APPROVED** [review]

File: Tasks/XamarinTestCloudV1/XamarinTestCloud.ps1 [hunk 2]
```diff
@@ -193,4 +193,11 @@ if($testCloudResults)
     Write-Host "##vso[task.addattachment type=Distributedtask.Core.Summary;name=Xamarin Test Cloud Results;]$mdReportFile"
 }
 
+$featureFlags = @{
+    failDeprecatedBuildTask  = [System.Convert]::ToBoolean($env:FAIL_DEPRECATED_BUILD_TASK)
+}
+if ($featureFlags.failDeprecatedBuildTask)
+{
+	throw "The XamarinTestCloud@1 (Xamarin Test Cloud) task has been deprecated since January 11, 2018 and will soon be retired. Use the AppCenterDistribute@3 task instead."
```
Comment (geekzter) [inline Tasks/XamarinTestCloudV1/XamarinTestCloud.ps1:201]: Shouldn't this string be taken from https://github.com/microsoft/azure-pipelines-tasks/blob/d4e470b1867696b700121b4170a2e1f2cbe2a077/Tasks/XamarinTestCloudV1/task.json#L30?
Now it is duplicated

File: _generated/XamarinTestCloudV1_Node20/XamarinTestCloud.ps1 [hunk 3]
```diff
@@ -193,4 +193,11 @@ if($testCloudResults)
     Write-Host "##vso[task.addattachment type=Distributedtask.Core.Summary;name=Xamarin Test Cloud Results;]$mdReportFile"
 }
 
+$featureFlags = @{
+    failDeprecatedBuildTask  = [System.Convert]::ToBoolean($env:FAIL_DEPRECATED_BUILD_TASK)
+}
+if ($featureFlags.failDeprecatedBuildTask)
+{
+	throw "The XamarinTestCloud@1 (Xamarin Test Cloud) task has been deprecated since January 11, 2018 and will soon be retired. Use the AppCenterDistribute@3 task instead."
```
Comment (geekzter) [inline _generated/XamarinTestCloudV1_Node20/XamarinTestCloud.ps1:201]: Shouldn't this be taken from task.json?

**Review by geekzter — CHANGES_REQUESTED** [review]

File: Tasks/XamarinTestCloudV1/XamarinTestCloud.ps1 (hunk 2 above)
Comment (Pranshu-Negi) [reply Tasks/XamarinTestCloudV1/XamarinTestCloud.ps1:201]: Same reason as here https://github.com/microsoft/azure-pipelines-tasks/pull/19471#discussion_r1467540461

**Review by geekzter — APPROVED** [review]

**Review by DergachevE — APPROVED** [review]

**Pull request merged**